network/Message.cpp

```diff
@@ -1309,3 +1318,16 @@ void ExtractAuthResponseMessageData(const Message& msg, std::string& player_name
 
 void ExtractSetAuthorizationRolesMessage(const Message &msg, Networking::AuthRoles& roles)
 { roles.SetText(msg.Text()); }
+
+void ExtractPlayerInfoMessageData(const Message &msg, std::map<int, PlayerInfo>& players) {
+    try {
+        std::istringstream is(msg.Text());
+        freeorion_xml_iarchive ia(is);
+        ia >> BOOST_SERIALIZATION_NVP(players);
+    } catch(const std::exception& err) {
+        ErrorLogger() << "ExtractPlayerInfo(const Message &msg, std::map<int, PlayerInfo>& players) failed!  Message:\n"
+                      << msg.Text() << "\n"
+                      << "Error: " << err.what();
+        throw;
```

Comment:
Maybe better to catch and log the error in the calling code instead or as well so that a serialization error doesn't crash the client

Reply:
Yes. It isn't critical message.